openlibrary/core/lending.py

```diff
@@ -111,6 +112,22 @@ def compose_ia_url(limit=None, page=1, subject=None, query=None, sorts=None, adv
     if subject:
         q += " AND openlibrary_subject:" + subject
 
+    if work_id:
+        work = web.ctx.site.get(work_id)
+        if work and _type.lower() in ["authors", "subjects"]:
+            if _type == "authors":
+                authors = []
+                for author in work.get_authors():
+                    authors.append(author.name)
+                    authors.append(','.join(author.name.split(' ', 1)[::-1]))
+                _q = ' OR '.join(['creator:(%s)' % author for author in authors])                
```

Comment:
Should there be quotation marks instead of `()`? Searching for `creator:(H. G. Wells)` is equivalent to `creator:"H." OR creator:"G." OR creator:"Wells"`, I think. Since we're handling the case of inverted author name, using quotes might yield better results.

Reply:
You can also make the resultant query a bit more succinct by using `OR` inside the `()`